activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -107,8 +107,9 @@ def _update_record(attribute_names = self.attribute_names)
             affected_rows
 
           # If something went wrong, revert the locking_column value.
-          rescue Exception
-            send(lock_col + "=", previous_lock_value.to_i)
+          rescue
```

Comment:
Don't need to apologize. it is important to try to remove code. This show we are missing test coverage for this.